### ISSUE TITLE
Investigate the "remote host identification has changed" warnings

### ISSUE THREAD

**Issue by adamwalach** [kyma-project/test-infra#3331]
**Description**

In prow jobs that provision VMs (to execute k3s or minikube) we often see warnings like this one:
```
Warning: Permanently added 'compute.4210891622161408321' (ECDSA) to the list of known hosts.
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@    WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
Someone could be eavesdropping on you right now (man-in-the-middle attack)!
It is also possible that a host key has just been changed.
The fingerprint for the ECDSA key sent by the remote host is
```

Full log: https://status.build.kyma-project.io/view/gcs/kyma-prow-logs/logs/post-master-kyma-integration/1364167461720559616

Since on every execution we provision a new VM, it's ssh key is not yet in the cache.
That might explain the warning, but we still need to investigate why it says "identification has changed".
That makes people nervous :).

**Expected result**

If this warning is a false positive it should be suppressed.

**Actual result**

```
IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
Someone could be eavesdropping on you right now (man-in-the-middle attack)!
```

**Steps to reproduce**

This log is present in every execution of `kyma-integtration-k3`:
https://status.build.kyma-project.io/?job=kyma-integration-k3s

**Comment by Halamix2** [comment]
It's possible to easily limit occurrences of this message to one, as seen in #3336:
First usage of `gcloud compute scp / ssh` generates new certificate and sends file/ executes command, displaying the aforementioned warning.
Removing `/root/.ssh/google_compute_known_hosts` file before next commands will force refresh of this file and will stop next warning messages.

This could be improved if we were able to generate and send ssh certificate before we call any `gcloud compute scp /ssh` commands

Perhaps `gcloud compute config-ssh` could help, but it would require additional permissions
```
 Updating project ssh metadata...
.............................................................................................................failed.
ERROR: (gcloud.compute.config-ssh) Could not add SSH key to project metadata:
 - Required 'compute.projects.setCommonInstanceMetadata' permission for 'projects/sap-kyma-prow-workloads' 
 ```

**Comment by adamwalach** [comment]
More info here: https://serverfault.com/a/992034
```
setting a scope, adding a service account, or even the lovely setting a description on an instance will force a replacement of a VM and the host key will change
```

**Comment by Halamix2** [comment]
Should be fixed in #3336
